### ISSUE TITLE
authkey periodicity in RSS

### ISSUE THREAD

**Issue by ghost** [WhatCD/Gazelle#118]
I am using a gazelle site and am subscribed to an RSS feed that works, but after a few weeks, the authkey in the feed changes (server side) and my feed connection breaks. Is there some way to change the feed url format, like for instance, including the password instead of the authkey that will make the feed continue to work without breaking?

**Comment by ghost** [comment]
@ReiGelado this is more of a Gazelle user question than a site-admin question. I don't run the site, so I can't modify the codebase. However, I'd love to have my RSS feed URL last longer than the 1-2 weeks that it currently does before the authkey expires. I was trying to figure out if I could customize the feed URL format to include, perhaps my password, so that it would not expire. Other similar solutions would solve my problem. But, contacting the siteops and asking them to change the authkey maximum age is probably not going to work for me.

**Comment by ReiGelado** [comment]
@triune 
Hi friend, sorry for the delay to respond.
Yes, if you do not have the auth, passkey and autkey it will not work so it will change if the sysops reconfigure the feed system.

**Comment by ghost** [comment]
@ReiGelado so are you saying the authkey will change for all users site-wide if the siteops change the feed system somehow? I am currently noticing that the authkey expires exactly every 14 days. This seems a little too periodic in nature to me such that it would be the result of a siteop feed change. It seems more likely that something is cycling the authkeys.

**Comment by ReiGelado** [comment]
I remembered, this key is set in the configuration file if it is changed it does this in the configuration file somehow the correct is you talk to your SysOP about it, it is set here:
```
// Keys
define('ENCKEY', ''); //Random key. The key for encryption
define('SITE_SALT', ''); //Random key. Default site wide salt for passwords, DO NOT LEAVE THIS BLANK/CHANGE AFTER LAUNCH!
define('SCHEDULE_KEY', ''); // Random key. This key must be the argument to schedule.php for the schedule to work.
define('RSS_HASH', ''); //Random key. Used for generating unique RSS auth key.
```

**Comment by ligh7s** [comment]
@ReiGelado You have no idea what you are talking about. Those are the private keys used to randomize user generated keys. Auth keys are generated for individual users and are cycled biweekly by the scheduler (8th and 22nd of each month, alongside invites).  
  
This repository is abandoned, and issue trackers aren't designed for user support. Post in the forums of the tracker you speak of instead. For a solution: you can set up a script to run an API call and substitute your new authkey into the RSS feed whenever it fails, assuming you know how to.

**Comment by ReiGelado** [comment]
@ligh7s 
I understood brother, had not yet studied the rss part of the site.
I tried to help;).
In which code snippet does it generate these new keys to use the rss? I searched the scheduler quickly and did not find it.

**Comment by ghost** [comment]
@ligh7s could you be a little more specific on the steps involved in your proposed solution? I was thinking I might have to write some python or javascript somewhere to get this to work, but I don't know where I can get the new authkey from. Is there a REST web service endpoint in Gazelle that I can authenticate (login/pass) and get the new authkey back as a response in JSON?

**Comment by ligh7s** [comment]
@triune https://github.com/WhatCD/Gazelle/wiki/JSON-API-Documentation#index contains your authkey.  
  
Post login credentials to https://site_na.me/login.php and create a session, then shoot a request to https://site_na.me/ajax.php?action=index. https://github.com/ligh7s/spread-red/blob/master/src/red.py#L55 is an example of one of my scripts logging in and storing the authkey.

**Comment by ghost** [comment]
Thanks for the info @ligh7s !! Definitely enough to get me going now!

**Comment by ligh7s** [comment]
Also most Gazelle RSS feeds support using the passkey as authentication over the authkey; is that an option for the site you're using?

**Comment by ghost** [comment]
@ligh7s not sure about that one... here's what a RSS feed looks like on the site:

https://site/feeds.php?
feed=torrents_notify_1234_abc12345678&
user=9999&
auth=abcdef12345678&
passkey=fsdjklafhjk3hkl24299999999999999&
authkey=999999999999999999999999aaaaaaaa&
name=feed_name

Looks like my site requires both?

**Comment by ReiGelado** [comment]
@triune 
https://xxxxxxx/feeds.php?feed=xxxxxx&user=xxxxxx&auth=xxxxxxx&passkey=xxxxxxxx&authkey=xxxxx

**Comment by jm134** [comment]
please tell me how to get unic code for "auth". it work fine for getting authkey
